src/main/java/com/example/demo/service/RecordService.java

```diff
@@ -19,9 +19,9 @@ public List<Record> getAllRecords() {
     }
 
     // Method to update the 'owned' field of a specific record
-    public Record updateOwned(int id, int owned) {
+    public Record updateOwned(int id, int howMuch, int balance) {
         Record record = recordRepository.findById(id).orElseThrow(() -> new RuntimeException("Record not found"));
-        record.setOwned(owned); // Update the owned field
+        record.buySome(howMuch); // Update the owned field
```

Comment:
It's not big at first but, remember not to code obvious lines of code - like `setOwned` method says it all... here with `buySome` (I'd still refactor the name) it's the same case - also maybe in the method itself there's more logic so this comment should not try to explain it all :)